9/README.md

```diff
@@ -0,0 +1,322 @@
+```
+
+* Values of DevVarLongStringArray and DevVarDoubleStringArray SHALL follow rules \<long-string-value\> 
+and \<double-string-value\>, respectively:
+```abnf
+ long-string-value = lvalue svalue 
+ double-string-value = dvalue svalue
+                       ; elements order implied by the above rules MAY be ignored
+```
+;where \<lvalue\> has type DevVarLongArray, \<dvalue\> has type DevVarDoubleArray and \<svalue\> has type 
+DevVarStringArray.
+
+#### State
+
+DevState type standardize a way of describing state of Device. Its values are from predefined set of 14 names.   
+
+* DevState name SHALL be according to the following rule:
+```abnf
+ DevState = "DevState"
+```
+
+* Value of DevState type SHALL follow the rule \<dev-state-value\>:
+```abnf
+ dev-state-value = "ALARM" / "INSERT" / "STANDBY" / "CLOSE" / "MOVING" / "UNKNOWN" / "DISABLE" / "OFF"
+ dev-state-value =/ "EXTRACT" / "ON" / "FAULT" / "OPEN" / "INIT" / "RUNNING"
+```
+
+#### Enumeration
+
+DevEnum data type allows to assign string Labels to DevShort values. It is valid only for Attributes. 
+
+* DevEnum name SHALL follow the following rule:
+```abnf
+ DevEnum = "DevEnum"
+```
+
+* Label values of DevEnum SHALL follow the rule \<devenum-label\>:
+```abnf
+ devenum-label = 1*VCHAR
+```  
+
+* DevEnum labels SHALL be unique in the context of an Attribute.
+* Each of DevEnum labels SHALL correspond to one and unique DevShort value. 
+* Related DevShort values SHALL be consecutive and SHALL start with 0. 
+* Values of DevEnum DataType SHALL be transported as values of DevShort.
+
+* * For any of Attribute of DevEnum data type, list of Labels SHALL be available as `enum_labels` Attribute property (see [4/Attribute](/4)).
+
+* Tango Controls SHALL allow a Tango Client to retrieve labels associated with the DevShort value.
+
+#### Pipe
+
+DevPipeBlob is a Data Type to transfer data related to Pipes (see 7/Pipe).
+
+* DevPipeBlob SHALL have the following name:
+```abnf
+ DevPipeBlob = "DevPipeBlob"
+```
+
+* A value of DevPipeBlob type SHALL follow the \<pipe-blob-value\> rule:
+
+```abnf
+ pipe-blob-value = name blob-data
+                   ; elements order implied by the above rule MAY be ignored
+
+ name = 1*VCHAR 
+
+ blob-data = *blob-data-element
+ 
+ blob-data-element = name value inner-blob inner-blob-name 
+                     ; elements order implied by the above rule MAY be ignored    
+
+ inner-blob = blob-data  
+```
+
+#### Exceptions 
+
+Results of failed operations (exceptions) within the Tango Controls are send as values/messages of DevFailed data type.
+
+* DevFailed type name is as follows:
+```abnf
+ DevFailed = "DevFailed"
+```
+
+* Any value of DevFiled type SHALL follow \<dev-failed-value\> rule:
+```abnf
+ dev-failed-value = 1*error 
+
+ error = reason severity desc origin
+
+ reason = *VCHAR ; SHALL be the reason of the failure
+ severity = "WARN" | "ERR" | "PANIC"
+ desc = *VCHAR ; SHOULD describe the failur in more details, eg. call stack
```

Comment:
The call stack is actually usually deduced from the origin fields of the **list** of error**s**.
For me, `origin `contains only 1 method name (The one from which the exception is thrown), this is not a call stack. But a call stack can be deducted from the origin fields of the list of errors.